File: frontend/src/types/staff.ts
// ========== STAFF TYPES ==========

export interface Staff {
  id: string;
  facility_id: string;
  first_name: string;
  last_name: string;
  dni: string | null;
  cuil: string | null;
  phone: string | null;
  email: string | null;
  position: string | null;
  specialty: string | null;
  license_number: string | null;
  hire_date: string | null;
  end_date: string | null;
  status: 'ACTIVE' | 'LEAVE' | 'INACTIVE';
  is_active: boolean;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface StaffCreate {
  facility_id: string;
  first_name: string;
  last_name: string;
  dni?: string;
  cuil?: string;
  phone?: string;
  email?: string;
  position?: string;
  specialty?: string;
  license_number?: string;
  hire_date?: string;
  notes?: string;
}

export interface StaffUpdate {
  first_name?: string;
  last_name?: string;
  dni?: string;
  cuil?: string;
  phone?: string;
  email?: string;
  position?: string;
  specialty?: string;
  license_number?: string;
  hire_date?: string;
  end_date?: string;
  status?: 'ACTIVE' | 'LEAVE' | 'INACTIVE';
  is_active?: boolean;
  notes?: string;
}

// ========== SHIFT TYPES ==========

export interface Shift {
  id: string;
  facility_id: string;
  name: string;
  start_time: string; // Format: "HH:MM:SS"
  end_time: string;
  color: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface ShiftCreate {
  facility_id: string;
  name: string;
  start_time: string;
  end_time: string;
  color?: string;
}

export interface ShiftUpdate {
  name?: string;
  start_time?: string;
  end_time?: string;
  color?: string;
  is_active?: boolean;
}

// ========== SHIFT ASSIGNMENT TYPES ==========

export interface ShiftAssignment {
  id: string;
  staff_id: string;
  shift_id: string;
  facility_id: string;
  date: string; // Format: "YYYY-MM-DD"
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface ShiftAssignmentCreate {
  staff_id: string;
  shift_id: string;
  facility_id: string;
  date: string;
  notes?: string;
}

export interface ShiftAssignmentUpdate {
  notes?: string;
}

export interface ShiftAssignmentWithDetails extends ShiftAssignment {
  staff_member: Staff;
  shift: Shift;
}

// ========== DASHBOARD TYPES ==========

export interface CurrentlyWorkingStaff {
  id: string;
  first_name: string;
  last_name: string;
  position: string | null;
  phone: string | null;
  shift_name: string;
  shift_start: string; // Format: "HH:MM:SS"
  shift_end: string;
  check_in_time: string | null;
  is_checked_in: boolean;
}

export interface StaffWorkloadSummary {
  staff_id: string;
  staff_name: string;
  total_shifts_month: number;
  total_hours_month: number;
  shifts_this_week: number;
}

export interface FacilityStaffDashboard {
  facility_id: string;
  facility_name: string;
  total_staff: number;
  active_staff: number;
  currently_working: CurrentlyWorkingStaff[];
  shifts_today: number;
  coverage_status: 'FULL' | 'UNDERSTAFFED' | 'OVERSTAFFED';
}

// ========== HELPER TYPES ==========

export type StaffPosition =
  | 'MEDICO'
  | 'ENFERMERO'
  | 'AUXILIAR'
  | 'ADMINISTRATIVO'
  | 'LIMPIEZA'
  | 'COCINA'
  | 'OTRO';

export const STAFF_POSITIONS: { value: StaffPosition | ''; label: string }[] = [
  { value: '', label: 'Seleccionar...' },
  { value: 'MEDICO', label: 'Médico' },
  { value: 'ENFERMERO', label: 'Enfermero/a' },
  { value: 'AUXILIAR', label: 'Auxiliar de Enfermería' },
  { value: 'ADMINISTRATIVO', label: 'Administrativo' },
  { value: 'LIMPIEZA', label: 'Personal de Limpieza' },
  { value: 'COCINA', label: 'Personal de Cocina' },
  { value: 'OTRO', label: 'Otro' },
];

export const STAFF_STATUS: { value: Staff['status'] | ''; label: string; color: string }[] = [
  { value: 'ACTIVE', label: 'Activo', color: 'green' },
  { value: 'LEAVE', label: 'Con Licencia', color: 'yellow' },
  { value: 'INACTIVE', label: 'Inactivo', color: 'gray' },
];

export const SHIFT_COLORS = [
  '#3B82F6', // Azul
  '#10B981', // Verde
  '#F59E0B', // Ámbar
  '#EF4444', // Rojo
  '#8B5CF6', // Púrpura
  '#EC4899', // Rosa
  '#14B8A6', // Teal
  '#F97316', // Naranja
];

File: backend/app/schemas/staff.py
from pydantic import BaseModel, Field
from uuid import UUID
from datetime import date, datetime, time
from typing import Optional, List


# ========== STAFF SCHEMAS ==========

class StaffCreate(BaseModel):
    facility_id: UUID
    first_name: str
    last_name: str
    dni: Optional[str] = None
    cuil: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    position: Optional[str] = None  # Cuidador, Enfermero, Médico, etc.
    specialty: Optional[str] = None
    license_number: Optional[str] = None
    hire_date: Optional[date] = None
    notes: Optional[str] = None


class StaffUpdate(BaseModel):
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    dni: Optional[str] = None
    cuil: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    position: Optional[str] = None
    specialty: Optional[str] = None
    license_number: Optional[str] = None
    hire_date: Optional[date] = None
    end_date: Optional[date] = None
    status: Optional[str] = Field(None, pattern="^(ACTIVE|LEAVE|INACTIVE)$")
    is_active: Optional[bool] = None
    notes: Optional[str] = None


class StaffResponse(BaseModel):
    id: UUID
    facility_id: UUID
    first_name: str
    last_name: str
    dni: Optional[str]
    cuil: Optional[str]
    phone: Optional[str]
    email: Optional[str]
    position: Optional[str]
    specialty: Optional[str]
    license_number: Optional[str]
    hire_date: Optional[date]
    end_date: Optional[date]
    status: str
    is_active: bool
    notes: Optional[str]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


# ========== SHIFT SCHEMAS ==========

class ShiftCreate(BaseModel):
    facility_id: UUID
    name: str
    start_time: time
    end_time: time
    color: Optional[str] = None


class ShiftUpdate(BaseModel):
    name: Optional[str] = None
    start_time: Optional[time] = None
    end_time: Optional[time] = None
    color: Optional[str] = None
    is_active: Optional[bool] = None


class ShiftResponse(BaseModel):
    id: UUID
    facility_id: UUID
    name: str
    start_time: time
    end_time: time
    color: Optional[str]
    is_active: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


# ========== SHIFT ASSIGNMENT SCHEMAS ==========

class ShiftAssignmentCreate(BaseModel):
    staff_id: UUID
    shift_id: UUID
    facility_id: UUID
    date: date
    notes: Optional[str] = None


class ShiftAssignmentUpdate(BaseModel):
    notes: Optional[str] = None


class ShiftAssignmentResponse(BaseModel):
    id: UUID
    staff_id: UUID
    shift_id: UUID
    facility_id: UUID
    date: date
    notes: Optional[str]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ShiftAssignmentWithDetails(ShiftAssignmentResponse):
    """Incluye detalles del staff y shift"""
    staff_member: StaffResponse
    shift: ShiftResponse

    class Config:
        from_attributes = True


# ========== DASHBOARD SCHEMAS ==========

class CurrentlyWorkingStaff(BaseModel):
    """Personal actualmente trabajando"""
    id: UUID
    first_name: str
    last_name: str
    position: Optional[str]
    phone: Optional[str]
    shift_name: str
    shift_start: time
    shift_end: time
    check_in_time: Optional[datetime]
    is_checked_in: bool


class StaffWorkloadSummary(BaseModel):
    """Resumen de carga laboral del personal"""
    staff_id: UUID
    staff_name: str
    total_shifts_month: int
    total_hours_month: int
    shifts_this_week: int


class FacilityStaffDashboard(BaseModel):
    """Dashboard de personal por facility"""
    facility_id: UUID
    facility_name: str
    total_staff: int
    active_staff: int
    currently_working: List[CurrentlyWorkingStaff]
    shifts_today: int
    coverage_status: str  # FULL, UNDERSTAFFED, OVERSTAFFED
